sql/txn.go

```diff
@@ -30,6 +34,15 @@ func (p *planner) BeginTransaction(n *parser.BeginTransaction) (planNode, error)
 // CommitTransaction commits a transaction.
 func (p *planner) CommitTransaction(n *parser.CommitTransaction) (planNode, error) {
 	err := p.txn.Commit()
+	if err != nil {
+		if restartErr, ok := err.(proto.TransactionRestartError); ok {
+			if restartErr.CanRestartTransaction() != proto.TransactionRestart_ABORT {
+				if replyErr := p.txn.Rollback(); replyErr != nil {
+					log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+				}
```

Comment:
This logic has been moved to `client.Txn.Commit`. It didn't seem easy to share with the logic in `client.Txn.exec`, but at least it is all in the same package now.